x/stake/endblock.go

```diff
@@ -252,3 +269,46 @@ func handleMatureUnbondingDelegations(k keeper.Keeper, ctx sdk.Context) ([]types
 
 	return completed, events
 }
+
+const (
+	maxProcessedRefundCount = 10
+)
+
+func handleRefundStake(ctx sdk.Context, k keeper.Keeper) sdk.Events {
+	iterator := k.IteratorAllDelegations(ctx)
+	defer iterator.Close()
+	var refundEvents sdk.Events
+	count := 0
+
+	bscValidators := k.GetAllValidators(ctx)
+	bscValidatorsMap := make(map[string]types.Validator, len(bscValidators))
+	for _, validator := range bscValidators {
+		bscValidatorsMap[validator.OperatorAddr.String()] = validator
+	}
+	boundDenom := k.BondDenom(ctx)
+
+	for ; iterator.Valid(); iterator.Next() {
+		delegation := types.MustUnmarshalDelegation(k.CDC(), iterator.Key(), iterator.Value())
+		validator := bscValidatorsMap[delegation.ValidatorAddr.String()]
+		amount := validator.TokensFromShares(delegation.GetShares()).RawInt()
+		if delegation.CrossStake {
+			ctx = ctx.WithCrossStake(true)
+		}
+		result := handleMsgSideChainUndelegate(ctx, types.MsgSideChainUndelegate{
+			DelegatorAddr: delegation.DelegatorAddr,
+			ValidatorAddr: delegation.ValidatorAddr,
+			Amount:        sdk.NewCoin(boundDenom, amount),
+			SideChainId:   k.ScKeeper.BscSideChainId(ctx),
+		}, k)
+		refundEvents = refundEvents.AppendEvents(result.Events)
+
+		count++
+		if count >= maxProcessedRefundCount {
+			break
+		}
+	}
+
+	_, ubdEvents := handleMatureUnbondingDelegations(k, ctx)
```

Comment:
I think we don't need this, this will be done by existing code.

Reply:
ok. removed